cpp/common/src/codingstandards/cpp/EncapsulatingFunctions.qll

```diff
@@ -18,6 +18,34 @@ class MainFunction extends MainLikeFunction {
   }
 }
 
+/**
+ * A test function from the GoogleTest infrastructure.
+ *
+ * Such functions can be treated as valid EntryPoint functions during analysis
+ * of "called" or "unused" functions. It is not straightforward to identify
+ * such functions, however, they have certain features that can be used for
+ * identification. This can be refined based on experiments/real-world use.
+ */
+class GoogleTestFunction extends MainLikeFunction {
+  GoogleTestFunction() {
+    // A GoogleTest function is named "TestBody" and
+    this.hasName("TestBody") and
```

Comment:
I think you could also mark the destructor as an "entry point", as it will also presumably be called by the gtest framework. That would eliminate the false positive you see in your test case.

Reply:
thanks for your suggestion .. let me try that out!

Reply:
It works! I've added SpecialMemberFunction instead of just the Destructor as entry points from GoogleTestFunction. This is addressed in [4b1429b](https://github.com/github/codeql-coding-standards/pull/725/commits/4b1429b6bf94bce541c32395eab0529a6d5ce9cb)